server/service/src/program/validate.rs

```diff
@@ -18,9 +18,14 @@ pub fn check_immunisation_program_exists(
 }
 pub fn check_program_name_exists(
     name: &str,
+    id: Option<String>,
     connection: &StorageConnection,
 ) -> Result<Option<ProgramRow>, RepositoryError> {
-    let program = ProgramRepository::new(connection)
-        .query_one(ProgramFilter::new().name(StringFilter::equal_to(name)))?;
+    let mut filter = ProgramFilter::new().name(StringFilter::equal_to(name));
```

Comment:
This is a cool way to make validate agnostic to insert vs update checks - I want to do the same to vaccine_courses as right now they're very verbose.